test/unit/test-navigation.js

```diff
@@ -634,6 +634,25 @@ describes.sandboxed('Navigation', {}, () => {
           expect(win.location.href).to.equal('https://www.pub.com/');
         });
 
+        it('should navigate relative to source url', () => {
+          win.location.href =
+            'https://cdn.ampproject.org/c/s/www.pub.com/dir/page.html';
+          const urlService = Services.urlForDoc(doc.documentElement);
+
+          sandbox.stub(urlService, 'getSourceUrl').callsFake(url => {
```

Comment:
shall we just install urlService to avoid this stub? that makes the test more robust to any future refactoring.

Reply:
I'm not sure I follow. `urlService` is already installed on the amp doc. I chose to stub the method because the internal anchor tag it uses is tied to the test runner's URL, not the fake window we'd expect.

Reply:
i thought it would read from the fake URL you just set in L638?